### ISSUE TITLE
Creating VS Solution Yields: "TypeError: get_host_target() missing 1 required positional argument: 'msvc_version':"

### ISSUE THREAD

**Issue by 19PHOBOSS98** [godotengine/godot#63709]
### Godot version

main

### System information

Win 10, AMD Ryzen 5 PRO 2400GE w/ Radeon Vega Graphics

### Issue description

I tried creating a Visual Studio Solution by following the compilation instructions:
https://docs.godotengine.org/en/stable/development/compiling/compiling_for_windows.html

using the following code:
`scons p=windows vsproj=yes`

but then this happens:
![image](https://user-images.githubusercontent.com/37253663/182008710-3497ae13-7679-4e2e-99ee-d00284049011.png)

I'm able to compile the project right using:
`scons -j8 platform=windows`
the bin file ran as expected. But I got stuck in prepping it for VS.
I tried reinstalling VS but that didn't work. Am I missing something?

What I've got installed so far:
+Visual Studio Community 2022 with both Desktop Development C++ and Game development with C++
+Python 3.10.15
+Git.2.37.1
+SCons: v4.4.0.

### Steps to reproduce

basically I followed this tutorial:
https://youtu.be/Ivs5lLcY5ug

**Comment by 19PHOBOSS98** [comment]
I figured out what was wrong

I looked at the methods.py file:
![image](https://user-images.githubusercontent.com/37253663/182009826-26760384-b756-4446-ba93-581a4b78e6cf.png)

the `get_host_target` function was missing an argument. I looked at the Scons Repo and found the function:
![image](https://user-images.githubusercontent.com/37253663/182010024-988bb4d3-9572-4d5a-a89e-3ad54c5b61e7.png)

...I tweaked the methods.py file a bit:
![image](https://user-images.githubusercontent.com/37253663/182010165-57ebed6c-52a6-48bb-9f39-05fced722290.png)


I got it to work:
![image](https://user-images.githubusercontent.com/37253663/182009790-edfd186b-b768-4ea9-a815-fc411bdf5914.png)

![image](https://user-images.githubusercontent.com/37253663/182010086-bcf16f83-d360-4976-bddd-9ca34101c9da.png)

![image](https://user-images.githubusercontent.com/37253663/182012888-23317cee-6183-4dc6-9a59-7e19bc0947a0.png)

Might be time to update this part of the file:
![image](https://user-images.githubusercontent.com/37253663/182013056-fd0b94f4-16b2-4554-9329-29c18dc9118f.png)
But I'm not so sure if changing this would be safe for anyone not using the latest version of SCons. 
Anyways, hope this helps :)

**Comment by akien-mga** [comment]
Thanks for the details. Sounds like SCons 4.4.0 added this new mandatory argument so we'll have to use either firm depending on which SCons version is used.
